nexusclient/client.go

```
@@ -0,0 +1,725 @@
+		return c.newUnexpectedResponseError(fmt.Sprintf("unexpected response status: %q", response.Status), response)
+	}
+	return nil
+}
+
+// Input for [nexusclient.Client.StartOperation].
+type StartOperationRequest struct {
+	// Name of the operation to start.
+	Operation string
+	// Callback URL to provide to the handle for receiving async operation completions. Optional.
+	CallbackURL string
+	// Request ID that may be used by the server handler to dedupe this start request.
+	// By default a v4 UUID will be generated by the client.
+	RequestID string
+	// Header to attach to the HTTP request. Optional.
+	Header http.Header
+	// Body of the operation request.
+	// If it is an [io.Closer], the body will be automatically closed by the client.
+	Body io.Reader
+}
+
+// NewJSONStartOperationRequest is shorthand for creating a StartOperationRequest with a JSON body.
+// Marhsals the provided value to JSON using [nexusapi.DefaultMarshaler].
+func NewJSONStartOperationRequest(operation string, v any) (request StartOperationRequest, err error) {
+	req := StartOperationRequest{}
+	var b []byte
+	b, err = nexusapi.DefaultMarshaler(v)
+	if err != nil {
+		return
+	}
+	req.Operation = operation
+	req.Header = http.Header{nexusapi.HeaderContentType: []string{nexusapi.ContentTypeJSON}}
+	req.Body = bytes.NewReader(b)
+	return
+}
+
+// StartOperation calls the configured Nexus endpoint to start an operation.
+// The operation may complete synchronously, delivering the result inline, or asynchronously returning a reference that
+// can be used via the returned handle to interact with the operation.
+//
+// Use the returned [OperationHandle] to retrieve the operation's result.
+func (c *Client) StartOperation(ctx context.Context, request StartOperationRequest) (*OperationHandle, error) {
+	if closer, ok := request.Body.(io.Closer); ok {
+		// Close the request body in case we error before sending the HTTP request.
+		defer closer.Close()
```

Hrmm, don't usually want to assume closers are safe for double close, but since we're discarding errors here, it's probably fine

---

Had the same thought here, and referred to the docs. It's fine since we ignore the error as you pointed out.
I'll clarify in a comment in code.